api/data.go

```diff
@@ -5,50 +5,53 @@ import (
 	"html/template"
 	"net/http"
 
-	"github.com/ONSdigital/dp-search-query/elasticsearch"
 	"github.com/ONSdigital/go-ns/log"
 )
 
-type dataParams struct {
+type dataLookupRequest struct {
 	Uris  []string
 	Types []string
 }
 
 var dataTemplates *template.Template
 
+// SetupTimeseries loads templates for use by the timeseries lookup handler and should be done only once
 func SetupData() error {
 	templates, err := template.ParseFiles("templates/data/queryByUri.tmpl")
 	dataTemplates = templates
 	return err
 }
 
-func DataLookupHandler(w http.ResponseWriter, req *http.Request) {
-	params := req.URL.Query()
-	templateParams := dataParams{
-		Uris:  params["uris"],
-		Types: params["types"],
-	}
-	var doc bytes.Buffer
+// DataLookupHandlerFunc returns a http handler function handling search api requests.
+func DataLookupHandlerFunc(elasticSearchClient ElasticSearcher) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		params := req.URL.Query()
+		reqParams := dataLookupRequest{
+			Uris:  params["uris"],
+			Types: params["types"],
+		}
+		var doc bytes.Buffer
 
-	err := dataTemplates.Execute(&doc, templateParams)
-	if err != nil {
-		log.Debug("Failed to create search from template", log.Data{"Error": err.Error(), "Params": templateParams})
-		http.Error(w, "Failed to create query", http.StatusInternalServerError)
-		return
-	}
-	formattedQuery, err := formatMultiQuery(doc.Bytes())
-	if err != nil {
-		log.Debug("Failed to format query for elasticsearch", log.Data{"Error": err.Error()})
-		http.Error(w, "Failed to create query", http.StatusInternalServerError)
-		return
-	}
-	responseData, err := elasticsearch.Search("", "", formattedQuery)
-	if err != nil {
-		log.Debug("Failed to query elasticsearch", log.Data{"Error": err.Error()})
-		http.Error(w, "Failed to run data query", http.StatusInternalServerError)
-		return
+		err := dataTemplates.Execute(&doc, reqParams)
+		if err != nil {
+			log.Debug("Failed to create search from template", log.Data{"Error": err.Error(), "Params": reqParams})
+			http.Error(w, "Failed to create query", http.StatusInternalServerError)
+			return
+		}
+		formattedQuery, err := formatMultiQuery(doc.Bytes())
+		if err != nil {
+			log.Debug("Failed to format query for elasticsearch", log.Data{"Error": err.Error()})
+			http.Error(w, "Failed to create query", http.StatusInternalServerError)
+			return
+		}
+		responseData, err := elasticSearchClient.Search("", "", formattedQuery)
+		if err != nil {
+			log.Debug("Failed to query elasticsearch", log.Data{"Error": err.Error()})
+			http.Error(w, "Failed to run data query", http.StatusInternalServerError)
+			return
+		}
+		dataWithResponse := "{\"responses\":[" + string(responseData) + "]}"
```

Comment:
I feel like this should really be a struct type and we should properly marshall the response data into it. Appreciate this is how it was before your PR but it feels a bit hacky to me. This could be completely invalid json - if that were the case `json.Marshal()` would return an error which we could use to log something useful and send the appropriate status.